src/Toolkit/Toolkit/UI/Controls/Legend/LegendEntry.cs

```diff
@@ -37,13 +41,75 @@ public class LegendEntry : object, ILayerContentItem
         public LegendEntry(object content)
         {
             Content = content;
+            if (content is INotifyPropertyChanged inpc)
+            {
+                inpc.PropertyChanged += Content_PropertyChanged;
+            }
+        }
+
+        /// <inheritdoc />
+        public event PropertyChangedEventHandler? PropertyChanged;
+
+        /// <summary>
+        /// Raises the <see cref="PropertyChanged"/> event.
+        /// </summary>
+        /// <param name="propertyName">name of the property changing</param>
+        protected virtual void OnPropertyChanged(string propertyName)
+        {
+            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
+        }
+
+        private void Content_PropertyChanged(object? sender, PropertyChangedEventArgs e)
+        {
+            if (e.PropertyName == nameof(ILayerContent.Name) || e.PropertyName == nameof(LegendInfo.Symbol))
+                PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(Content)));
+            else if (e.PropertyName == nameof(LegendInfo.Symbol))
+                PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(Symbol)));
```

Comment:
Is the logic in this if-else correct? It seems that the "else" is unreachable -- a PCE for Symbol will never fire.

Reply:
Nice catch. None of objects ones we listen to have a property called `Content` 🤦